wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -337,40 +380,48 @@ instance Arbitrary EstimateFeesError where
     arbitrary = EstFeesTxCreationFailed <$> arbitrary
 
 estimateFees :: ActiveWallet
-             -> PassPhrase
-             -- ^ The spending password.
              -> CoinSelectionOptions
              -- ^ The options describing how to tune the coin selection.
              -> HdAccountId
              -- ^ The source HD Account from where the payment should originate
              -> NonEmpty (Address, Coin)
              -- ^ The payees
              -> IO (Either EstimateFeesError Coin)
-estimateFees activeWallet@ActiveWallet{..} spendingPassword options accountId payees = do
-    res <- newTransaction activeWallet spendingPassword options accountId payees
+estimateFees activeWallet@ActiveWallet{..} options accountId payees = do
+    res <- newUnsignedTransaction activeWallet options accountId payees
     case res of
          Left e  -> return . Left . EstFeesTxCreationFailed $ e
-         Right (tx, _txMeta, originalUtxo) ->
-             -- calculate the fee as the difference between inputs and outputs.
+         Right (_db, tx, originalUtxo) -> do
+             let change = utxChange tx
+             -- calculate the fee as the difference between inputs and outputs. The
+             -- final 'sumOfOutputs' must be augmented by the change, which we have
+             -- available in the 'UnsignedTx' as a '[Coin]'.
+             --
              -- NOTE(adn) In case of 'SenderPaysFee' is practice there might be a slightly
              -- increase of the projected fee in the case we are forced to pick "yet another input"
              -- to be able to pay the fee, which would, in turn, also increase the fee due to
              -- the extra input being picked.
              return $ Right
-                    $ sumOfInputs tx originalUtxo `unsafeSubCoin` sumOfOutputs tx
+                    $ sumOfInputs tx originalUtxo
+                    `unsafeSubCoin`
+                    (repeatedly Core.unsafeAddCoin change (sumOfOutputs tx))
   where
-      -- Unlike a block, a /single transaction/ cannot have inputs that sum to
-      -- more than maxCoinVal
-      sumOfInputs :: TxAux -> Utxo -> Coin
-      sumOfInputs tx utxo =
-          let inputs = Set.fromList $ toList . _txInputs . taTx $ tx
-          in unsafeIntegerToCoin $
-               utxoBalance (utxo `utxoRestrictToInputs` inputs)
-
-      sumOfOutputs :: TxAux -> Coin
-      sumOfOutputs tx =
-          let outputs = _txOutputs . taTx $ tx
-          in paymentAmount outputs
+    -- Tribute to @edsko
+    repeatedly :: (a -> b -> b) -> ([a] -> b -> b)
+    repeatedly = flip . foldl' . flip
+
+    -- Unlike a block, a /single transaction/ cannot have inputs that sum to
+    -- more than maxCoinVal
+    sumOfInputs :: UnsignedTx -> Utxo -> Coin
+    sumOfInputs tx utxo =
+        let inputs = Set.fromList $ map fst . toList . utxOwnedInputs $ tx
+        in unsafeIntegerToCoin $
+             utxoBalance (utxo `utxoRestrictToInputs` inputs)
```

Comment:
We don't need to rely on the UTxO, since we have _resolved_ inputs (can just add the outputs).